test/e2e/e2e_test.go

```diff
@@ -0,0 +1,41 @@
+package e2e_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("e2e", func() {
+
+	var device EdgeDevice
+	var err error
+
+	BeforeEach(func() {
+		device, err = NewEdgeDevice("edgedevice1")
+		Expect(err).To(BeNil())
+	})
+
+	AfterEach(func() {
+		deviceId, err := device.GetId()
+		Expect(err).To(BeNil())
```

Comment:
Nitpick: On AfterEAch we shouldn't run assertions. If for any reason cannot get the DeviceId, it'll stop and it'll be not `unregistered` or `remove()`


It shouldn't happen, you should be able to always get the device in this case, but I'll take extra care in these situations :-)

Reply:
And I remembered to set up something like this on Cilium, we do the checks on JustAfterEach, and now it's part of Ginkgo 2.0, maybe we should update the version. 

https://onsi.github.io/ginkgo/#separating-diagnostics-collection-and-teardown-justaftereach



My approach will be: 



- JustAfterEach: Check race-conditions, validate device is still in there. 
- AfterEach: Force removing all devices, etc..



Also, with the new version you can also do AfterAll() BeforeAll(), so it'll be cool for these cases. :-)

Reply:
Unfortunately we need to migrate all of the tests before we can use Ginkgo 2 here. Let's have it as a separate story, maybe?